providers/alibaba/src/airflow/providers/alibaba/cloud/hooks/oss.py

```diff
@@ -27,7 +27,11 @@
 from oss2.exceptions import ClientError
 
 from airflow.exceptions import AirflowException
-from airflow.hooks.base import BaseHook
+
+try:
+    from airflow.sdk import BaseHook
+except ImportError:
+    from airflow.hooks.base import BaseHook  # type: ignore
```

Comment:
```suggestion
try:
    from airflow.sdk import BaseHook
except ImportError:
    from airflow.hooks.base import BaseHook  # type: ignore[no-redef]
```

Reply:
Maybe worth putting it under (including moving `BaseOperator` there):

providers/alibaba/src/airflow/providers/alibaba/version_compat.py

Reply:
Yep, will investigate here: https://github.com/apache/airflow/issues/52676